geopandas/geodataframe.py

```diff
@@ -84,7 +84,7 @@ def __init__(self, *args, **kwargs):
                     hasattr(self["geometry"].values, "crs")
                     and self["geometry"].values.crs
                     and crs
-                    and self["geometry"].values.crs != crs
+                    and self["geometry"].values.crs != CRS.from_user_input(crs)
```

Comment:
I find it strange that this is needed, since this is basically the first thing pyproj does when checking for equality ...

Reply:
Yeah, I was surprised as well. I just tried to figure out what is going on and it seems like a pyproj issue (cc @snowman2).

```py
crs = CRS(27700)

crs == 27700
True

crs != 27700  # this is strange
True

crs != CRS(27700)
False
```

Reply:
It seems that pyproj implements `__eq__`, but not `__neq__`. I am not sure what Python falls back to then.

Reply:
I'll do `not self["geometry"].values.crs == crs` here.

Reply:
Based on https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq, it seems that Python should negate `==` when `__ne__` is not defined (at least starting with Python 3). But which is clearly not happening in the above example. In any case, adding it to pyproj seems best. Will open an issue there.

Reply:
https://github.com/pyproj4/pyproj/pull/596